discovery-provider/src/tasks/entity_manager/user.py

```diff
@@ -0,0 +1,134 @@
+import logging
+from typing import Dict
+
+from src.models.users.user import User
+from src.tasks.entity_manager.user_replica_set import parse_sp_ids
+from src.tasks.entity_manager.utils import (
+    USER_ID_OFFSET,
+    Action,
+    EntityType,
+    ManageEntityParameters,
+    copy_user_record,
+)
+from src.tasks.user_replica_set import get_endpoint_string_from_sp_ids
+from src.tasks.users import (
+    update_legacy_user_images,
+    update_user_metadata,
+    validate_user_record,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def is_valid_user_tx(params: ManageEntityParameters):
+    user_id = params.user_id
+
+    if params.entity_type != EntityType.USER:
+        raise Exception("Invalid User Transaction, wrong entity type")
+
+    if params.action == Action.CREATE:
+        if user_id in params.existing_records[EntityType.USER]:
+            raise Exception("Invalid User Transaction, user does not exist")
+        if user_id < USER_ID_OFFSET:
+            raise Exception("Invalid User Transaction, user id offset incorrect")
+    else:
+        # update / delete specific validations
+        if user_id not in params.existing_records[EntityType.USER]:
+            raise Exception("Invalid User Transaction, user does not exist")
+        wallet = params.existing_records[EntityType.USER][user_id].wallet
+        if wallet and wallet.lower() != params.signer.lower():
+            raise Exception(
+                "Invalid User Transaction, user wallet signer does not match"
+            )
+    return True
+
+
+def update_user_record(params: ManageEntityParameters, user: User, metadata: Dict):
+    update_user_metadata(
+        params.session, params.redis, user, metadata, params.web3, params.challenge_bus
+    )
+    user.metadata_multihash = params.metadata_cid
+    user = update_legacy_user_images(user)
+    user = validate_user_record(user)
+    return user
+
+
+def create_user(params: ManageEntityParameters):
+    if not is_valid_user_tx(params):
```

Comment:
nit this never returns false (only True or throws an exception). maybe change it to a void and `validate_user_tx` so it only throws an exception.